src/modules/AmidoBuild/exported/Invoke-Helm.ps1

```diff
@@ -76,34 +83,69 @@ function Invoke-Helm() {
         $chartpath,
 
         [string]
-        $releasename,
         # Name of the release
+        $releasename,
 
         [string]
-        $namespace
         # Namespace to deploy the release into
+        $namespace,
+
+        [string]
+        $repositoryName,
+
+        [string]
+        $repositoryUrl
 
     )
 
     # Define parameter checking vars
     $missing = @()
+    $checkParams = @()
+
+    switch ($PSCmdlet.ParameterSetName) {
+        "install" {
+            # Check that some arguments have been set
+            $checkParams = @("provider", "target", "identifier", "namespace", "releasename","namespace")
+        }
 
+        "repo" {
+            $checkParams = @("repositoryName", "repositoryUrl")
+        }
+
+        "custom" {
+            $checkParams = @("arguments")
```

Comment:
Custom currently has to log in, so required should also include "provider", "target", "identifier" too, and both install and custom need the "k8sauthrequired"